lib/inits/migrations/setupMigrations.ts

```diff
@@ -1,69 +1,70 @@
-import { setupActions } from "../../actions/setupActions"
 import { EmVer } from "../../emverLite/mod"
 import { SDKManifest } from "../../manifest/ManifestTypes"
 import { ExpectedExports } from "../../types"
 import { once } from "../../util/once"
 import { Migration } from "./Migration"
 
-export class Migrations {
+export class Migrations<WD> {
   private constructor(
     readonly manifest: SDKManifest,
-    readonly migrations: Array<Migration<any>>,
+    readonly migrations: Array<Migration<WD, any>>,
   ) {}
   private sortedMigrations = once(() => {
-    const migrationsAsVersions = (this.migrations as Array<Migration<any>>).map(
-      (x) => [EmVer.parse(x.options.version), x] as const,
-    )
+    const migrationsAsVersions = (
+      this.migrations as Array<Migration<WD, any>>
+    ).map((x) => [EmVer.parse(x.options.version), x] as const)
     migrationsAsVersions.sort((a, b) => a[0].compareForSort(b[0]))
     return migrationsAsVersions
   })
   private currentVersion = once(() => EmVer.parse(this.manifest.version))
-  static of<Migrations extends Array<Migration<any>>>(
+  static of<Migrations extends Array<Migration<WD, any>>, WD>(
     manifest: SDKManifest,
-    ...migrations: EnsureUniqueId<Migrations>
+    ...migrations: EnsureUniqueId<Migrations, WD>
   ) {
-    return new Migrations(manifest, migrations as Array<Migration<any>>)
+    return new Migrations(manifest, migrations as Array<Migration<WD, any>>)
   }
   async init({
     effects,
+    utils,
     previousVersion,
-  }: Parameters<ExpectedExports.init>[0]) {
+  }: Parameters<ExpectedExports.init<WD>>[0]) {
```

Comment:
See types expected exports will not have nor know about utils, therefore don't.
Move the creation of utils into these two fn's